pinc/DPDatabase.inc

```diff
@@ -15,6 +15,7 @@ final class DPDatabase
     private static $_default_db_charset = null;
     private static $_default_db_collation = null;
     public static $skip_encoding_check = false;
+    public static $tracing_log = null;
```

Comment:
Nitpick, but could we get into the habit of adding type annotations?
```public static ?string $tracing_log = null;```

Reply:
I've added a chaser commit that adds type defs to all class variables and function definitions. Might as well do it all at once. Sandbox has been updated.

We use all of these all over the place so phpstan + smoketest + unit tests + "does loading pages work" should be sufficient.